code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -295,3 +295,22 @@
 	icon_state = "pill8"
 	list_reagents = list(/datum/reagent/iron = 30)
 	rename_with_volume = TRUE
+	
+/obj/item/reagent_containers/pill/floor_pill
+	name = "☼Floor Pill☼"
+	desc = "This is a masterful floor pill. It is decorated with ashwalker bone and encircled with bands of iron and silver. The object menaces with spikes of wood. On the item is an image of an assistant eating a floor pill. The assistant is dying. On the item is an image of four medical doctors. The medical doctors are praying."
+	icon_state = "pill8"
+	list_reagents = list()
+	
+/obj/item/reagent_containers/pill/floor_pill/Initialize()
+	var/list/picked_reagents = list()
+	var/list/all_reagents = subtypesof(/datum/reagent/)
+	var/reagent_count = rand(1,all_reagents.len)
+	for(var/i in 1 to reagent_count)
+		picked_reagents += pick_n_take(all_reagents)
+	if(picked_reagents)
+		volume = (picked_reagents.len * 10)
+	for(var/picked_reagent in picked_reagents)
+		list_reagents[picked_reagent] = 10
```

Comment:
```suggestion
	var/reagent_volume = 0
	if(picked_reagents)
		volume = min(picked_reagents.len * 10, 1000)
		reagent_volume = volume / picked_reagents.len
	
	for(var/picked_reagent in picked_reagents)
		list_reagents[picked_reagent] = reagent_volume
```

This way the reagents transferred aren't limited by the stomach capacity of the mob eating them unless they've eaten something else immediately before.

Reply:
Thanks!